extensions/3DTILES_batch_table_hierarchy/README.md

```diff
@@ -0,0 +1,426 @@
+
+A feature's `batchId` is used to access its `classId` and `parentCount`. Therefore, the values in the `classIds` and `parentCounts` arrays are initially ordered by `batchId` and followed by non-feature instances.
+
+The `parentCounts` and `parentIds` arrays form an instance hierarchy. A feature's properties include those defined by its own class and any properties from ancestor instances.
+
+In some cases multiple ancestors may share the same property name. This can occur if two ancestors are the same class or are different classes with the same property names. For example, if every class defined the property "id", then it would be an overloaded property. In such cases it is up to the implementation to decide which value to return.
+
+Finally, `classIds`, `parentCounts`, and `parentIds` may instead be references to data in the binary body. If omitted, `componentType` defaults to `UNSIGNED_SHORT`. `type` is implicitly `SCALAR`.
+
+```json
+"classIds" : {
+    "byteOffset" : 0,
+    "componentType" : "UNSIGNED_SHORT"
+};
+```
+
+JSON schema definitions can be found in [3DTILES_batch_table_hierarchy.json](./schema/3DTILES_batch_table_hierarchy.json).
+
+## Examples
+
+### Feature classes
+
+Going back to the example of a parking lot with car, lamp post, and tree features, a Batch Table might look like this:
+
+```json
+{
+  "extensions" : {
+    "3DTILES_batch_table_hierarchy" : {
+      "classes" : [
+        {
+          "name" : "Lamp",
+          "length" : 3,
+          "instances" : {
+            "lampStrength" : [10, 5, 7],
+            "lampColor" : ["yellow", "white", "white"]
+          }
+        },
+        {
+          "name" : "Car",
+          "length" : 3,
+          "instances" : {
+            "carType" : ["truck", "bus", "sedan"],
+            "carColor" : ["green", "blue", "red"]
+          }
+        },
+        {
+          "name" : "Tree",
+          "length" : 2,
+          "instances" : {
+            "treeHeight" : [10, 15],
+            "treeAge" : [5, 8]
+          }
+        }
+      ],
+      "instancesLength" : 8,
+      "classIds" : [0, 0, 0, 1, 1, 1, 2, 2]
+    }
+  }
+}
+```
+
+Since this example does not contain any sort of hierarchy, the `parentCounts` and `parentIds` are not included, and `instancesLength` just equals the tile's `batchLength`.
+
+A `classId` of 0 indicates a "Lamp" instance, 1 indicates a "Car" instance, and 2 indicates a "Tree" instance.
+
+A feature's `batchId` is used to access its class in the `classIds` array. Features with a `batchId` of 0, 1, 2 are "Lamp" instances, features with a `batchId` of 3, 4, 5 are "Car" instances, and features with `batchId` of 6 and 7 are "Tree" instances.
+
+The feature with `batchId = 5` is the third "Car" instance, and its properties are
+
+```
+carType : "sedan"
+carColor : "red"
+```
+
+Batch Table Hierarchy, parking lot:
+
+![batch table hierarchy parking lot](figures/batch-table-hierarchy-parking-lot.png)
```

Comment:
Oh good catch about the pictures!

The json formatting is a little off on this picture. The other picture looks good.